### DISCUSSION TITLE
[bug?] App keeps restaring when using sea_orm

### DISCUSSION THREAD

**Discussion by Jasenkoo** [tauri-apps/tauri#4075] [Q&A]
Hello everyone, I am generally new to the Rust language and in process of learning, however, I am not sure if this is an issue or if something is missing. Using Vite and Vue for the client. On the backend, I want to connect rust to SQLite.

There is nothing much going on, my **main.rs** looks like this:

```rs
use sea_orm::Database;

#[tokio::main]
async fn main() {
    let db = Database::connect("sqlite://test.db?mode=rwc")
        .await
        .unwrap();

    println!("{:?}\n", db);

  
    tauri::Builder::default()
        .invoke_handler(tauri::generate_handler![])
        .run(tauri::generate_context!())
        .expect("error while running tauri application");
}
```

After I start the app with `tauri dev` the app compiles fine, `printlln!` prints `SqlxSqlitePoolConnection`, however, it keeps restarting, it opens `WebView` for a second, and then close and compiles again and again.

In between some compilations, I can see this warning if it matters something, this is the console output

```sh
Compiling app v0.1.0 (D:\Projects\tauri-example\src-tauri)
    Finished dev [unoptimized + debuginfo] target(s) in 2.99s
     Running `target\debug\app.exe`
SqlxSqlitePoolConnection

warning: Error finalizing incremental compilation session directory `\\?\D:\Projects\tauri-example\src-tauri\target\debug\incremental\app-1mt7itu6mci1i\s-g9kbtq3uv9-1lyav1v-working`: Access is denied. (os error 5)

Compiling app v0.1.0 (D:\Projects\tauri-example\src-tauri)
    Finished dev [unoptimized + debuginfo] target(s) in 4.93s
     Running `target\debug\app.exe`
SqlxSqlitePoolConnection

Compiling app v0.1.0 (D:\Projects\tauri-example\src-tauri)
    Finished dev [unoptimized + debuginfo] target(s) in 3.00s
     Running `target\debug\app.exe`
SqlxSqlitePoolConnection

```

As you can see it just compiles non-stop. When I remove the line where it connects to the database it is working fine. So is this something from tauri side or sea_orm, since it is not throwing errors or not compiling, just restarting. Any help is appreciated.

**Comment by FabianLars** [comment]
Can you try updating rust? Also maybe whitelist the project dir in your antivirus software. (And cleaning the target dir may be a good idea too)

The first one is because there were many incremental compilation bugs in prior versions and the latter is because someone on discord had a similar problem caused by Avira

**Reply by Jasenkoo** [reply]
Tnx, will try

**Reply by Jasenkoo** [reply]
It didn't help, rust was already on the latest version, I don't use antivirus except for windows defender (tried disabling controlled folder access but didn't help either), and deleted target dir. Will see to choose some other async runtime, or change the library. I was already playing with diesel and rusqlite and both work without issues.

**Reply by Jasenkoo** [reply]
Some additional info, after investigating a little, although didn't notice at first, after the application exits, there are 2 additional files added in the src folder where is my local db.

- `test.db-shm`
- `test.db-wal`

They are keeping it from starting properly again. If removed the app complies again without issues.

**Reply by silentEAG** [reply]
Hi, friend, I meet the same problem. I think the reason may be `tauri dev` will watch project fold for changes. the files sea-orm generate may occur to change so make the app restaring. using `sqlite://../test.db` works fine.

**Reply by Jasenkoo** [reply]
@silentEAG When exiting the app with `ctrl` + `c` seems to cause it but when the close button is clicked there is no problem.

**Comment by ZaneH** [comment]
If anyone has the same issue, you'll need the `TAURI_DEV_WATCHER_IGNORE_FILE` environment variable set to an absolute path to your `.taurignore` file. You'll also need to specific which files to stop watching. This `.taurignore` example unwatches everything except the `./src` and `./src-tauri` folders

```
*

!src/**/*
!src-tauri/**/*
```
